python/lsst/pipe/analysis/visitAnalysis.py

```diff
@@ -231,109 +236,101 @@ def runDataRef(self, dataRefList, tract=None):
-            if self.config.doPlotMags:
-                self.plotMags(catalog, filenamer, repoInfo.dataId, butler=repoInfo.butler,
-                              camera=repoInfo.camera, ccdList=ccdListPerTract, hscRun=repoInfo.hscRun,
-                              zpLabel=self.zpLabel)
-            if self.config.doPlotStarGalaxy:
-                if "ext_shapeHSM_HsmSourceMoments_xx" in catalog.schema:
-                    self.plotStarGal(catalog, filenamer, repoInfo.dataId, butler=repoInfo.butler,
-                                     camera=repoInfo.camera, ccdList=ccdListPerTract, hscRun=repoInfo.hscRun,
-                                     zpLabel=self.zpLabel)
-                else:
-                    self.log.warn("Cannot run plotStarGal: " +
-                                  "ext_shapeHSM_HsmSourceMoments_xx not in catalog.schema")
-            if self.config.doPlotSizes:
-                if "base_SdssShape_psf_xx" in catalog.schema:
-                    self.plotSizes(catalog, filenamer, repoInfo.dataId, butler=repoInfo.butler,
-                                   camera=repoInfo.camera, ccdList=ccdListPerTract, hscRun=repoInfo.hscRun,
-                                   zpLabel=self.zpLabel)
-                else:
-                    self.log.warn("Cannot run plotSizes: base_SdssShape_psf_xx not in catalog.schema")
-            if self.config.doPlotCentroids and self.haveFpCoords:
-                self.plotCentroidXY(catalog, filenamer, repoInfo.dataId, butler=repoInfo.butler,
-                                    camera=repoInfo.camera, ccdList=ccdListPerTract, hscRun=repoInfo.hscRun,
-                                    zpLabel=self.zpLabel)
+
+                # Dict of all parameters common to plot* functions
+                plotKwargs = dict(butler=repoInfo.butler, camera=repoInfo.camera, ccdList=ccdListPerTract,
+                                  hscRun=repoInfo.hscRun, zpLabel=self.zpLabel)
+
+                if self.config.doPlotFootprintNpix:
+                    self.plotFootprintHist(catalog,
+                                           filenamer(repoInfo.dataId, description="footNpix", style="hist"),
+                                           repoInfo.dataId, **plotKwargs)
+                    self.plotFootprint(catalog, filenamer, repoInfo.dataId, plotRunStats=False,
+                                       highlightList=[("parent", 0, "yellow"), ], **plotKwargs)
+
+                if self.config.doPlotQuiver:
+                    self.plotQuiver(catalog,
+                                    filenamer(repoInfo.dataId, description="ellipResids", style="quiver"),
+                                    dataId=repoInfo.dataId, scale=2, **plotKwargs)
+
+                # Create mag comparison plots using common ZP
+                if self.config.doPlotMags and not commonZpDone:
+                    zpLabel = "common (" + str(self.config.analysis.commonZp) + ")"
```

Comment:
Reading your code, it can be hard sometimes to tell what variables are sequences vs. scalars.  Also, `zpLabel = "common (%s)" % self.config.analysis.commonZp` would work fine.